drivers/system_pathways/app/models/system_pathways/chart_base.rb

```diff
@@ -141,13 +144,84 @@ def average(value, count)
       scope.merge(race_scope)
     end
 
+    private def filter_for_race_with_details_filter(scope, race_filter)
+      return scope.where(id: multi_racial_clients(scope).select(:id)) if race_filter.races.include?('MultiRacial')
+
+      query = races.except('MultiRacial', 'RaceNone').keys.map { |k| [k.underscore.to_sym, false] }.to_h
+      race_filter.races.each do |column|
+        next if column.in?(['MultiRacial', 'RaceNone'])
+
+        query[column.underscore.to_sym] = true
+      end
+      scope.where(query)
+    end
+
+    private def filter_for_ethnicity_with_details_filter(scope, filter)
+      unknown_query = races.except('MultiRacial', 'RaceNone').keys.map { |k| [k.underscore.to_sym, false] }.to_h
+
+      # These are mutually exclusive, so, although the filter is an array, ignore multiples
+      case filter.ethnicities.first
+      when :hispanic_latinaeo
+        scope.where(hispanic_latinaeo: true)
+      when :non_hispanic_latinaeo
+        scope.where(hispanic_latinaeo: false).
+          where.not(unknown_query).except(:hispanic_latinaeo) # Must have at least one race response
+      when :unknown
+        scope.where(unknown_query)
+      end
+    end
+
+    private def filter_for_race_and_ethnicity_with_details_filter(scope, filter)
+      base_query = races.except('MultiRacial', 'RaceNone').keys.map { |k| [k.underscore.to_sym, false] }.to_h
+      combination = filter.race_ethnicity_combinations.first
+
+      query = case combination
+      when :am_ind_ak_native
+        base_query.merge(am_ind_ak_native: true)
+      when :am_ind_ak_native_hispanic_latinaeo
+        base_query.merge(am_ind_ak_native: true, hispanic_latinaeo: true)
```

Comment:
Would there be any utility/cleanup/DRY/ease of reading if this were a hash?  Something like:
```
combinations = {
  am_ind_ak_native: {am_ind_ak_native: true},
  am_ind_ak_native_hispanic_latinaeo: {am_ind_ak_native: true, hispanic_latinaeo: true),
  asian: {asian: true},
  asian_hispanic_latinaeo: {asian: true, hispanic_latinaeo: true},
...
}
query = base_query.mege(combinations[combination])
```
Multi-racial would need some special casing, but otherwise i think it would work.  
OR
We could even build the hash from an array of the fields with a  simple map:
```
combinations = {} 
[:am_ind_ak_native, :asian].map do |race|
  combinations[race] = base_query.merge(race: true)
combinations["#{race}_#{hispanic_latinaeo}".to_sym] = base_query.merge(race: true, hispanic_latinaeo: true)
end
combinations[:multi_racial] = { hispanic_latinaeo: false }
...
```

Reply:
I don't think so as I think the `case` is very explicit, and I think that would move the logic away from its use, but, I could easily be convinced as I'm almost always in favor of declarative code!